bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -145,7 +140,48 @@
             }
         }
 
-        classModelBuilder.instanceCreatorFactory(new InstanceCreatorFactoryImpl<T>(new CreatorExecutable<T>(clazz, noArgsConstructor)));
+        classModelBuilder.instanceCreatorFactory(new InstanceCreatorFactoryImpl<T>(
+                new CreatorExecutable<T>(clazz, noArgsConstructor, false)));
+    }
+
+    static <T> void configureRecordModelBuilder(final ClassModelBuilder<T> classModelBuilder, final Class<T> clazz) {
+        Map<String, TypeParameterMap> propertyTypeParameterMap = new HashMap<>();
+        String declaringClassName = clazz.getSimpleName();
+        Map<String, PropertyMetadata<?>> propertyNameMap = new HashMap<>();
+
+        List<String> genericTypeNames = getGenericTypeNames(clazz);
+
+        RecordComponent[] recordComponents = clazz.getRecordComponents();
+        Class<?>[] types = new Class<?>[recordComponents.length];
+        for (int i = 0; i < recordComponents.length; i++) {
+            RecordComponent recordComponent = recordComponents[i];
+            String name = recordComponent.getName();
+            types[i] = recordComponent.getType();
+
+            Method getter;
+            try {
+                getter = clazz.getDeclaredMethod(name);
+            } catch (NoSuchMethodException e) {
+                throw new AssertionError("Record must have getter method with same name as field");
```

Comment:
I wasn't sure what to put in the catch block. These calls should never fail because the methods are guaranteed to to exist with records.

Reply:
Generally we seem to use a CodecConfiguration exception for pretty much all exceptions in Codecs. I think an IllegalStateException would be fine here and below. Also pass the  NoSuchMethodException exception on as well.

Reply:
Got it, done.